scripts/generate-conda-envs.py

```diff
@@ -57,13 +60,22 @@ def conda(self) -> Reqs:
         if self.ctk_version == "none":
             return ()
 
-        return (
+        deps = (
             f"cudatoolkit={self.ctk_version}",  # runtime
             "cutensor>=1.3.3",  # runtime
             "nccl",  # runtime
             "pynvml",  # tests
         )
 
+        # gcc 11.3 is incompatible with nvcc <= 11.5.
+        if self.compilers and (V(self.ctk_version) <= V("11.5")):
```

Comment:
I think we also want to do this only on linux (which is presumably where gcc will be used as the backing for the c-compiler and cxx-compiler packages).

Reply:
I keep forgetting we are nominally supporting non-linux platforms :)

Reply:
@manopapad on OSX `self.ctk_version == "none"` so all this is skipped anyway. Do you still want an explicit check for linux? I guess that might future proof if windows is added.

Reply:
Yeah, let's add it for future-proofing.

Reply:
https://github.com/nv-legate/legate.core/pull/612/commits/6f0d6d0a130ec908e23778ccb8582afb49934414